translib/app_interface.go

```diff
@@ -84,16 +85,17 @@ type appInterface interface {
 	translateDelete(d *db.DB) ([]db.WatchKeys, error)
 	translateGet(dbs [db.MaxDB]*db.DB) error
 	translateAction(dbs [db.MaxDB]*db.DB) error
-	translateSubscribe(dbs [db.MaxDB]*db.DB, path string) (*notificationOpts, *notificationInfo, error)
+	translateSubscribe(req *translateSubRequest) (*translateSubResponse, error)
 	processCreate(d *db.DB) (SetResponse, error)
 	processUpdate(d *db.DB) (SetResponse, error)
 	processReplace(d *db.DB) (SetResponse, error)
 	processDelete(d *db.DB) (SetResponse, error)
 	processGet(dbs [db.MaxDB]*db.DB) (GetResponse, error)
 	processAction(dbs [db.MaxDB]*db.DB) (ActionResponse, error)
+	processSubscribe(req *processSubRequest) (processSubResponse, error)
 }
 
-//App modules will use this function to register with App interface during boot up
+// App modules will use this function to register with App interface during boot up
 func register(path string, info *appInfo) error {
 	var err error
 	log.Info("Registering for path =", path)
```

Comment:
While correcting this file, it would be helpful to convert stuff like this to a more explicit call like `log.Infof("Registering for path =%s", path)`

Reply:
Are you suggesting to use Infof() instead of Info() ??
I am not performing any such cleanup of the existing code in this PR. I am only introducing translateSubscribe() & peocessSubscribe() methods and types used by them. All other changes you see (like line #98 here) are due to gofmt.